British republican flag selection

Danarca opened the pull request:
5 different flags for a republican Great Britain, saw them here: https://www.reddit.com/r/vexillology/comments/16tlvyl/the_most_popular_flags_of_the_british_republican/

Some are fictional, but I added them because of the _Rule of Cool_.
I made them in code instead of the usual .tga /.dds emblem shenaniganery, without the British coat of arms the post specified. I felt they detract from the flags.

The selection event is in an on_monthly `events = {` section, but the trigger for the event itself specifies that the country must be Great Britain, have one of the 2 democratic republic laws (presidential and parliamentary), and not have the flag variant variable.

Upon firing, the variant variable is immediately set to 0 (which doesn't trigger a flag change), ensuring that the event only fires once.

Each option changes the variable to numbers from 1 - 5, with each number representing a flag.
And speaking of the flag triggers, the flags requires the country to be Great Britain (naturally), for the flag variant variable to be set to their own values, and for them to be a republic (but not communist or anarchist, as those have their own flags). I'm pretty sure fascism has its law set through corporate_state, but could be wrong..?

I hope Brock is happy with the flag choices 😄

Danarca committed:
5 british republican flags

Danarca committed:
Republicans flag wont be used for communists or anarchists

Danarca committed:
Selection event, confirmed to work

Just need an event image now!

Danarca committed:
Background for event added + simpler loc

Danarca committed:
Localisation fiddling

Danarca committed:
Reformatted all files to common sense

Danarca committed:
Missing underscore

Danarca committed:
Merge branch 'Imperia-british-republican-flags' of https://github.com/Imperia-Development-Victoria-3/Victoria-3-Dev into Imperia-british-republican-flags

HiddeLekanne commented on common/on_actions/imperia_on_monthtly_pulse.txt:
```diff
@@ -447,6 +447,7 @@
 	}
 	events = {
 		infamy_rebalance.1	#updates provinces each month
+		british_republican_flag_events.1	# British republican flag event, should fire only once.
```
Why is this not a decision?

Danarca replied:
I could do that, just figured this would be a one-time thing.

I *could* make it an event, if you think that would really improve things?

Danarca replied:
After mulling over it, I found a decision-solution I'm happy with :)
Every time Britain goes republican, or changes their republic form between parliamentary/presidential, they get a chance to change the flag, done by decision. This is done on an `on_law_activated` on_action, which also works if Britain is force-republicanised through events.

The guys at VTM figured out that `on_law_activated` has `root` as the law, NOT the country (as the documentation claims), which must've been a headache for them. But thanks to that I could do it this way :)

Quote from their on_action list; 

> `# root = ??????????????????`
> `# DOCUMENTATION SAYS COUNTRY BUT ITS LAW `
> `# WHY DO YOU LIE????????????`
> `# WHY???`

So yea, smells like adventure lmao

Danarca replied:
Also, I see that Vic3-tiger thought the documentation was correct, so we also need to make exceptions for the scope in that on_action. 
Everything works ingame as it should, and only as specified in the on_action. So this is correct.

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-british-republican-flags

Danarca committed:
Moved flag change to event, get the opportunity to change every time you go republican or change republic type

Credits must be given to the VTM Spain author, who wrote this in their on_law_activated documentation;
# root = ??????????????????
# DOCUMENTATION SAYS COUNTRY BUT ITS LAW
# WHY DO YOU LIE????????????
# WHY???

Without that I would've been lost. o7

Danarca committed:
Reformatted all files to common sense

Danarca committed:
Added systemic errors to our suppression list
